pyocd/target/builtin/target_STM32H723xx.py

```diff
@@ -0,0 +1,317 @@
+# pyOCD debugger
+# Copyright (c) 2020 Bartek Wolowiec
+# Copyright (c) 2021 Chris Reed
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import logging
+from ...coresight.coresight_target import CoreSightTarget
+from ...core.memory_map import (FlashRegion, RamRegion, MemoryMap)
+from ...coresight.cortex_m import CortexM
+from ...coresight import dap,ap
+
+LOG = logging.getLogger(__name__)
+
+class DBGMCU:
+
+    # Via APB-ap (AP2)
+    #BASE =   0xe00e1000
+    # via AHB-ap (AP0,1)
+    BASE =   0x5c001000
+
+    IDC =  BASE + 0x000
+    CR =   BASE + 0x004
+    CR_VALUE = (0x3f | # keep running in stop sleep and standby
+               0x07 << 20 | # enable all debug components
+               0x07
+               )
+
+    ABP3 = BASE + 0x034
+
+class FlashPeripheral:
+    def __init__(self):
+        self.flashaddr = 0x2000+0x12000000+0x40000000
+        self.flash_keyr = self.flashaddr + 4
+        self.flash_optkeyr = self.flashaddr + 8
+        self.flash_optcr = self.flashaddr + 0x18
+        self.flash_cr = self.flashaddr + 0xc
+        self.flash_sr = self.flashaddr + 0x10
+        self.flash_optsr_cur = self.flashaddr + 0x1c
+        self.flash_optsr_prg = self.flashaddr + 0x20
+
+
+class MiniAP(object):
```

Comment:
Can this `MiniAP` class be moved to a common location so there is only one copy of the code? Maybe `pyocd.coresight.minimal_mem_ap.py`?

A common implementation should mask the `IDR` with 0x0fffe00f and check against 0x04770001. This will verify it's an Arm AHB-AP of any variant or revision.

Reply:
Yeah, i will write it, but it i am not in a position to test with the F767.

Reply:
Well, me either. 😄 But I can verify it by inspection and hand testing.

Reply:
I aded it like suggested, and tested on the two H7 devices. For the F7xx, it turns out that the exact MiniAP from the F767 worked on the H7 devices as well. So i used that. But I did add the `&` like you suggested.

Reply:
Thanks. The mask just ensures it will work on different and future devices without needing any changes.